### PR TITLE
Adcm-2014 change xpath to css in locators

### PR DESCRIPTION
**Pull request by annademina**

### CODE REVIEW

File: tests/ui_tests/app/page/common/base_page.py [hunk 1]
```diff
@@ -299,21 +276,46 @@ def assert_page_is_opened():
         with allure.step(f'Wait page {page_name} is opened'):
             wait_until_step_succeeds(assert_page_is_opened, period=0.5, timeout=timeout)
 
-    def set_locator_value(self, locator: Locator, value: str) -> None:
-        """Fill locator with value."""
+    def send_text_to_element(
+        self, locator: Locator, text: str, is_with_clear: bool = True, timeout: Optional[int] = None
```
Comment (Sealwing) [inline tests/ui_tests/app/page/common/base_page.py:280]: `is_with_clear` looks heavy, maybe just `clean_input`?
and what's the motivation to make it a flag and not decide on a test level either we want to clean input or not?

**Review by madlabman — DISMISSED** [review]

File: tests/ui_tests/app/page/common/configuration/page.py [hunk 2]
```diff
@@ -167,7 +167,17 @@ def fill_password_and_confirm_fields(self, password: str, confirmation: str, dis
     @allure.step('Click on group {title}')
     def click_on_group(self, title: str):
         """Click on group with given title"""
-        self.find_and_click(self.locators.group_btn(title))
+
+        def check_group_expanded(group: WebElement):
+            return "expanded" in group.get_attribute("class")
```
Comment (Sealwing) [inline tests/ui_tests/app/page/common/configuration/page.py:172]: You name it "check", but no check in body, so better to name it `is_group_expanded` or even `is_expanded_in_group_class`

File: tests/ui_tests/app/page/common/configuration/page.py [hunk 3]
```diff
@@ -167,7 +167,17 @@ def fill_password_and_confirm_fields(self, password: str, confirmation: str, dis
...
+
+        def click_group():
+            group = self.find_element(self.locators.group_btn(title))
+            is_expanded = check_group_expanded(group)
+            group.click()
+            assert check_group_expanded(self.find_element(self.locators.group_btn(title))) != is_expanded
```
Comment (Sealwing) [inline tests/ui_tests/app/page/common/configuration/page.py:178]: Add assertion message, please

and maybe make things happening here move verbose: method not only clicks on a group, but it also expects its "status" to change, so maybe add it to the method's description or smt like that

File: tests/ui_tests/app/page/common/configuration/page.py [hunk 4]
```diff
@@ -224,16 +234,19 @@ def check_config_fields_visibility(
         assert len(visible_fields) == 0, f"Those fields should be visible: {visible_fields}"
 
     @contextmanager
-    def wait_rows_change(self):
+    def wait_rows_change(self, amount_to_change: Optional[int] = None):
```
Comment (Sealwing) [inline tests/ui_tests/app/page/common/configuration/page.py:237]: `amount_to_change` looks like `expected_rows_amount` or `amount_after_change`

File: tests/ui_tests/app/page/common/base_page.py (hunk 1 above)
Comment (annademina) [reply tests/ui_tests/app/page/common/base_page.py:280]: changed

File: tests/ui_tests/app/page/common/configuration/page.py (hunk 2 above)
Comment (annademina) [reply tests/ui_tests/app/page/common/configuration/page.py:172]: changed

File: tests/ui_tests/app/page/common/configuration/page.py (hunk 4 above)
Comment (skhomuti) [reply tests/ui_tests/app/page/common/configuration/page.py:237]: `expected_rows` looks good

File: tests/ui_tests/app/page/common/configuration/page.py (hunk 3 above)
Comment (annademina) [reply tests/ui_tests/app/page/common/configuration/page.py:178]: added
waiting for "status" to change is necessary to check if the group is really opened or closed, it's just a helper

File: tests/ui_tests/app/page/common/configuration/page.py (hunk 4 above)
Comment (annademina) [reply tests/ui_tests/app/page/common/configuration/page.py:237]: changed

File: tests/ui_tests/app/page/common/base_page.py [hunk 5]
```diff
@@ -293,8 +294,9 @@ def send_text_to_element(
         expected_value = element.get_property('value') + text
 
         def send_keys_and_check():
-            if is_with_clear:
-                self.clear_by_keys(locator)
+            if clean_input:
+                with allure.step("Clear element"):
+                    self.clear_by_keys(locator)
```
Comment (skhomuti) [inline tests/ui_tests/app/page/common/base_page.py:299]: this is already a step

**Review by skhomuti — CHANGES_REQUESTED** [review]

File: tests/ui_tests/app/page/common/base_page.py (hunk 5 above)
Comment (annademina) [reply tests/ui_tests/app/page/common/base_page.py]: removed

**Review by skhomuti — APPROVED** [review]

**Review by Sealwing — APPROVED** [review]

**Review by Sealwing — APPROVED** [review]

**Review by Sealwing — APPROVED** [review]

**Review by Sealwing — APPROVED** [review]

**Pull request merged**